_tasks/sync-proposal-data.mjs

```diff
@@ -0,0 +1,346 @@
+const markdown = require('remark-parse');
+const html = require('remark-html');
+const table2json = require('tabletojson').Tabletojson;
+const log = require('fancylog');
+const he = require('he');
+
+const DIR_DATA = './_data';
+const DIR_TEMP = './tmp';
+
+rmdirSync(DIR_TEMP, { recursive: true });
+mkdirSync(DIR_TEMP);
+
+const ghFileImporter = new GhFileImporter(DIR_TEMP);
+const rawPrpslReadmeUrl =
+  'https://raw.githubusercontent.com/tc39/proposals/master/README.md';
+
+await ghFileImporter.importFileFromUrl(rawPrpslReadmeUrl);
+const proposalsReadmeMarkdown = readFileSync(
+    `${DIR_TEMP}${separator}README.md`,
+    'utf-8'
+  );
+log.debug(proposalsReadmeMarkdown);
+
+/** @enum {string} */
+const UnicodeEscapes = {
+  leftDoubleQuotes: '\u201c', // “
+  rightDoubleQuotes: '\u201d', // ”
+  errorSymbol: '\u24E7', // ⓧ
+  warningSymbol: '\u26A0', // ⚠
+};
+
+// -----------------------------------------------------------------------------
+// Events
+// -----------------------------------------------------------------------------
+
+process.on('uncaughtException', (err /* , origin */) => {
+  const errorText = `Uncaught ${
+    err instanceof TypeError ? 'TypeError' : 'Error'
+  }: ${err.message}`;
+
+  writeSync(process.stderr.fd, UnicodeEscapes.errorSymbol + ' ' + errorText +
+            newlineMarker);
+});
+
+// -----------------------------------------------------------------------------
+// Typedefs
+// -----------------------------------------------------------------------------
+
+/**
+ * @typedef {{
+ *   date: string,
+ *   url: string,
+ * }}
+ */
+const PresentedObj = {};
+
+// -----------------------------------------------------------------------------
+// Helpers
+// -----------------------------------------------------------------------------
+
+/**
+ * Gets the URL value from the Markdown link key in the proposals README.
+ * @param {string} urlKeyLike
+ * @returns {string}
+ */
+function getUrlFromDoc(urlKeyLike) {
+  // log.debug({ urlKeyLike });
+  const urlKeyLikeRegex = new RegExp(`\\[${urlKeyLike}.*\\]: (.*)`, 'g');
+  // log.debug({ urlKeyLikeRegex });
+  const results = urlKeyLikeRegex.exec(proposalsReadmeMarkdown);
+  // log.debug({ results });
+  return results[1];
+}
+
+/**
+ * Downloads the proposal's README file using the GitHub API, parses the
+ * Markdown file, and returns the last code block.
+ * @param {string} prpslId
+ * @returns {Array<string>}
+ */
+async function getCodeSample(prpslId) {
+  const codeBlocks = [];
+  try {
+    // Some proposal repos use a different branch for the README.md file (i.e.,
+    // `gh-pages`). Because of this, we need to get some repo metadata before
+    // knowing what the URL is.
+    let rawReadmeUrl;
+    await ghFileImporter
+      .fetchPathMetadata('tc39', prpslId, 'README.md')
+      .then(function (res) {
+        // log.info(res.data.download_url);
+        rawReadmeUrl = res.data.download_url;
+      })
+      .catch(function (err) {
+        throw err;
```

Comment:
this catch is now redundant, and can just be omitted

Reply:
Hmm, I originally put them there to avoid seeing Node's unhandled promise rejection warnings.

Reply:
if you await the chain, you'll never get a warning from the chain (but you might still get one from the containing async function).

A catch that's just a rethrow is always a noop.